packages/block-library/src/navigation/create-data-tree.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Creates a nested, hierarchical tree representation from unstructured data that
+ * has an inherent relationship defined between individual items.
+ *
+ * For example, by default, each element in the dataset should have an `id` and
+ * `parent` property where the `parent` property indicates a relationship between
+ * the current item and another item with a matching `id` propertys.
+ *
+ * This is useful for building linked lists of data from flat data structures.
+ *
+ * @param {Array} dataset linked data to be rearranged into a hierarchical tree based on relational fields.
+ * @param {string} id the property which uniquey identifies each entry within the array.
+ * @param {*} relation the property which identifies how the current item is related to other items in the data (if at all).
+ */
+function createDataTree( dataset, id = 'id', relation = 'parent' ) {
+	const hashTable = Object.create( null );
```

Comment:
What does `Object.create` do that `{}` doesn't?

Reply:
I think I did this because I wanted to create an object without a prototype reference. I guess in reality it doesn't hugely matter, but I was taught if you're using an object as a hash lookup you should make sure it's totally empty (so to speak).

Reply:
TIL 😄